Firestore/core/src/firebase/firestore/util/status.h

```diff
@@ -136,6 +136,21 @@ extern std::string StatusCheckOpHelperOutOfLine(const Status& v,
   FIREBASE_DEV_ASSERT_MESSAGE_WITH_EXPRESSION( \
       val.ok(), val.ok(), StatusCheckOpHelperOutOfLine(val, #val).c_str())
 
+// Run a command that returns a util::Status.  If the called code returns an
+// error status, return that status up out of this method too.
+//
+// Example:
+//   RETURN_IF_ERROR(DoThings(4));
+//
+// (Shamelessly stolen from
+// https://github.com/google/protobuf/blob/master/src/google/protobuf/stubs/status_macros.h)
+#define RETURN_IF_ERROR(expr)                                                \
```

Comment:
I'm against it, but if you're strongly in favor, I'll defer this to Gil.

I find it less readable due to SHOUTY NATURE of macros, especially when a lambda is passed to the `operation`. If it's possible to disable clang-format selectively, then the if statement could be a one-liner `if (!s.ok()) return;`

Also, if `Writer` didn't return status, but stored it inside, then these returns would be unnecessary.

Reply:
Some thoughts:
  * I'd prefer to avoid macro magic generally, and especially those that obscure control flow.
  * We already have clang-format configured to allow `if (!s.ok()) return s;`.
  * Accumulating the error in the Writer makes this considerably simpler.